src/api/java/de/fearnixx/jeak/service/controller/ResponseEntity.java

```diff
@@ -32,12 +39,61 @@ public void removeHeader(String header) {
     }
 
     @Override
-    public void setResponseEntity(T responseEntity) {
-        this.responseEntity = responseEntity;
+    public void setEntity(T entity) {
+        this.entity = entity;
+    }
+
+    @Override
+    public T getEntity() {
+        return entity;
     }
 
     @Override
-    public T getResponseEntity() {
-        return responseEntity;
+    public void setStatus(int status) {
+        this.httpStatus = status;
+    }
+
+    @Override
+    public int getStatus() {
+        return httpStatus;
+    }
+
+    public static class Builder<T> {
+        private Map<String, String> headers;
+        private T responseEntity;
+        private int httpStatus;
+
+        public Builder(T responseEntity) {
+            this.responseEntity = responseEntity;
+            this.headers = new HashMap<>();
+        }
+
+        public Builder() {
+            this.responseEntity = null;
+            this.headers = new HashMap<>();
+        }
+
+        public Builder<T> withHeader(Map.Entry<String, String> header) {
```

Comment:
This method signature is oddly specific. Is there a reason we must support ``Map.Entry`` for ``#withHeaders`` when we already have a method accepting generic ``Map``s?

Reply:
I thought it to maybe be comfortable to have the option to just pass entries of a map, maybe while streaming without having the need of splitting them. Of course I could also just remove it and add it again, when someone actually needs it (that sounds good to me).

Reply:
Sounds reasonable enough. But I would think that most streams are either ``(key, value) ->`` or on either the entry or value set. Therefore, I would consider a stream of ``Map.Entry`` to be an edge case.

Reply:
So I will throw it and if the requirements change, we can always add it again.